src/libAtomVM/nifs.c

```diff
@@ -3014,33 +3014,16 @@ static term nif_binary_last_1(Context *ctx, int argc, term argv[])
 static term nif_binary_part_3(Context *ctx, int argc, term argv[])
```

Comment:
I don't think this is necessary: BIFs have always precedence over NIFs, so the NIF version can just be dropped.
By the way, how did you notice this change? Which OTP version introduced it?

Reply:
_I think_ it was used in guard somewhere and it was failing with "can't call function in guard" but my memory is fuzzy.

Reply:
Could we please have a test that asserts this behavior as a BIF, typically using an expression such as:

```
    B = ?MODULE:id(<<"foo">>),
    ok = if binary_part(B, {2, 1}) =:= <<"o">> -> ok; true -> not_ok end.
```

The code above definitely crashes AtomVM 0.6 and I understand your PR fixes that.

Reply:
> Could we please have a test that asserts this behavior as a BIF, typically using an expression such as:
> 
> ```
>     B = ?MODULE:id(<<"foo">>),
>     ok = if binary_part(B, {2, 1}) =:= <<"o">> -> ok; true -> not_ok end.
> ```
> 
> The code above definitely crashes AtomVM 0.6 and I understand your PR fixes that.

Sounds like a good idea rebasing it on top of release-0.6.

Reply:
Ok, some context went missing and I apologize for my wrong review. This is not just about changing an existing `erlang:binary_part/3` NIF to BIF, this is about adding `erlang:binary_part/3` BIF while there is still a `binary:part/3` that is sill here.